src/plugins/data_source/config.ts

```diff
@@ -61,4 +61,11 @@ export const configSchema = schema.object({
   }),
 });
 
-export type DataSourcePluginConfigType = TypeOf<typeof configSchema>;
+export type ConfigSchema = TypeOf<typeof configSchema>;
+
+export type DataSourcePluginConfigType = ConfigSchema & {
+  globalOpenSearchConfig: {
+    requestTimeout: Duration;
+    pingTimeout: Duration;
+  };
+};
```

Comment:
nit: Not related to this change, maybe we should add datasource own request timeout